libbpf-tools/tcpconnect.h

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2020 Anton Protopopov
+#ifndef __TCPCONNECT_H
+#define __TCPCONNECT_H
+
+/* The maximum number of items in maps */
+#define MAX_ENTRIES 8192
+
+/* The maximum number of ports to filter */
+#define MAX_PORTS 64
+
+#define TASK_COMM_LEN 16
+
+struct ipv4_flow_key {
+	__u32 saddr;
+	__u32 daddr;
+	__u16 dport;
+};
+
+struct ipv6_flow_key {
+	unsigned __int128 saddr;
+	unsigned __int128 daddr;
+	__u16 dport;
+};
+
+struct event {
+	int af; // AF_INET or AF_INET6
+	__u64 ts_us;
```

Comment:
4 byte padding here, you can move uid or pid up here. __int128s below have 16 byte alignment, so better move them up in front to optimize layout

Reply:
Moved bigger fields to the top